tasks.py

```diff
@@ -15,263 +15,269 @@
+    build_image(context, name, python_ver, image_ver)
 
 
 @task
-def pytest(context, name=NAME, python_ver=PYTHON_VER):
+def pytest(context, name=NAME, image_ver=IMAGE_VER, local=INVOKE_LOCAL):
     """This will run pytest for the specified name and Python version.
 
     Args:
         context (obj): Used to run specific commands
         name (str): Used to name the docker image
-        python_ver (str): Will use the Python version docker image to build from
+        image_ver (str): Will use the container version docker image
+        local (bool): Define as `True` to execute locally
     """
     # pty is set to true to properly run the docker commands due to the invocation process of docker
     # https://docs.pyinvoke.org/en/latest/api/runners.html - Search for pty for more information
     # Install python module
-    docker = f"docker run -it -v {PWD}:/local {name}-{python_ver}:latest"
-    context.run(
-        f"{docker} /bin/bash -c 'poetry install && pytest "
-        "--cov=diffsync --cov-config pyproject.toml --cov-report html --cov-report term -vv'",
-        pty=True,
-    )
+    exec_cmd = "pytest -vv"
+    run_cmd(context, exec_cmd, name, image_ver, local)
 
 
 @task
-def black(context, name=NAME, python_ver=PYTHON_VER):
+def black(context, name=NAME, image_ver=IMAGE_VER, local=INVOKE_LOCAL):
     """This will run black to check that Python files adherence to black standards.
 
     Args:
         context (obj): Used to run specific commands
         name (str): Used to name the docker image
-        python_ver (str): Will use the Python version docker image to build from
+        image_ver (str): Define image version
+        local (bool): Define as `True` to execute locally
     """
     # pty is set to true to properly run the docker commands due to the invocation process of docker
     # https://docs.pyinvoke.org/en/latest/api/runners.html - Search for pty for more information
-    docker = f"docker run -it -v {PWD}:/local {name}-{python_ver}:latest"
-    context.run(f"{docker} black --check --diff .", pty=True)
+    exec_cmd = "black --check --diff ."
+    run_cmd(context, exec_cmd, name, image_ver, local)
 
 
 @task
-def flake8(context, name=NAME, python_ver=PYTHON_VER):
+def flake8(context, name=NAME, image_ver=IMAGE_VER, local=INVOKE_LOCAL):
     """This will run flake8 for the specified name and Python version.
 
     Args:
         context (obj): Used to run specific commands
         name (str): Used to name the docker image
-        python_ver (str): Will use the Python version docker image to build from
-    """
-    # pty is set to true to properly run the docker commands due to the invocation process of docker
-    # https://docs.pyinvoke.org/en/latest/api/runners.html - Search for pty for more information
-    docker = f"docker run -it -v {PWD}:/local {name}-{python_ver}:latest"
-    context.run(f"{docker} flake8 .", pty=True)
-
-
-@task
-def mypy(context, name=NAME, python_ver=PYTHON_VER):
```

Comment:
Looks like the `mypy` task got removed - please add it back.

Reply:
good catch, I added it back